x/staking/types/expected_keepers.go

```diff
@@ -61,7 +61,7 @@ type ValidatorSet interface {
 	StakingTokenSupply(sdk.Context) math.Int                     // total staking token supply
 
 	// slash the validator and delegators of the validator, specifying offence height, offence power, and slash fraction
-	Slash(sdk.Context, sdk.ConsAddress, int64, int64, sdk.Dec) math.Int
+	Slash(sdk.Context, sdk.ConsAddress, int64, int64, sdk.Dec, Infraction)
 	SlashWithInfractionReason(sdk.Context, sdk.ConsAddress, int64, int64, sdk.Dec, Infraction) math.Int
```

Comment:
So, v0.45.x-ics did have a breaking change in Slash, but in v0.47 we decided not to break the API for users, hence the `SlashWithInfractionReason`.

Reply:
@vuong177 hey can you check this out, can we move the slash func for consumer keeper to SlashWithInfractionReason